### DISCUSSION TITLE
Why not just use GT_BBOX

### DISCUSSION THREAD

**Discussion by leijue222** [fredzzhang/spatio-attentive-graphs#81] [Q&A] [answered]
Hi, @fredzzhang . There're some ideas I want to discuss with you.
Why not just use GT_BOX, but use the detected BBOX+GT_BBOX?
https://github.com/fredzzhang/spatially-conditioned-graphs/blob/a44093ea0d77e0d4a27b1694c18df2fec9af5770/interaction_head.py#L93-L110

I'm thinking without a detector. Assuming that the GT_BBOX coordinates of people and objects are known, the detector is not required to detect the coordinates. Then go directly to the HOI task.

**Comment by fredzzhang** [comment, marked as answer]
Hi @leijue222,

The ground truth boxes and detected boxes are data of the same modality but from different distributions. Ideally, training and test data should be drawn from the same distribution. This is why people usually do a random split to get training and test sets—to remove sampling bias.

Therefore, since at test time we only have access to the detected boxes, we should technically only use detected boxes during training as well. However, since the detector is not perfect, there is a chance it will miss certain objects, especially those from rare object classes, making the interactions including those objects not learnable. As a result, the combination of ground truth and detected boxes are used.

Fred.

**Reply by leijue222** [reply]
I can understand why two types of boxes are used in the training phase now. 
But why can't we just use GT_BOX when testing? (Since the VCOCO test set we know GT_BBOX.)

**Reply by fredzzhang** [reply]
Well this is a detection task in the first place, so you are not supposed to have access to the ground truth boxes during testing. You can still report the performance using the ground truth boxes as a diagnostic measure. But those results won't be comparable to existing numbers.

**Comment by leijue222** [comment]
![1671502099670](https://user-images.githubusercontent.com/30276789/208564339-1512ed48-217e-4d19-a770-84a23b56aa0c.png)
I conducted further experiments, removed resize, and only used GT BOX for training.
Look the table, only use GT_BBOX during training, use (GT_BBOX+Detected_BBOX) or only (GT_BBOX) during evaluation.
That's weird, I don't understand why evaluating with just GT_BOX results in so much worse metrics.

**Reply by fredzzhang** [reply]
That does seem a bit odd. I'm not quite sure why this is the case. Most likely there is a bug somewhere in the code. You will have to dig deeper into the cached detections.
